.rubocop.yml

```diff
@@ -18,6 +18,12 @@ AllCops:
     - 'truffleruby/target/classes/**/*'
     - 'mxbuild/**/*'
 
+Lint/AssignmentInCondition:
+  Description: Don't use assignment in conditions.
+  StyleGuide: "#safe-assignment-in-condition"
+  Enabled: true
+  AllowSafeAssignment: true
+
```

Comment:
> It improves consistency since it's a more strict rule.

Well, no, the code before was perfectly consistent in never using `(=)`.
I still feel `(=)` is ugly and everything but idiomatic Ruby.

Assignment in loops is a perfectly nice thing and cannot be replaced by any other construct without extra `break`, so I think this cop is really just making the code worse for no gain.
```ruby
while line = io.gets # or item = queue.pop for another common example
  yield line
end
```

Removing assignments in conditionals is also problematic if they are multiple branches:
```ruby
if str = try_convert(arg, :to_str)
  ...
elsif hash = try_convert(arg, :to_h)
  ...
end
```
would become
```ruby
str = try_convert(arg, :to_str)
if str
  ...
else
  hash = try_convert(arg, :to_h)
  if hash
    ...
  end
end
```
which I think we all agree looks worse.

Can we add the cops we all agree on and revisit this one and string literals once we have some experience with the existing cops?